### DISCUSSION TITLE
Error updating to 3.24 or 3.25.1

### DISCUSSION THREAD

**Discussion by codigoergosum** [PHP-CS-Fixer/PHP-CS-Fixer#7277] [Q&A]
I use PHP-CS-Fixer through the Junstyle MS VS Code plugin, running on Windows 10.  Having found an issue with a fixer feature, I tried updating to the newest version, but this failed horribly.  After running the self updater from the php-cs-fixer.phar held within the Junstyle plugin, I get "Unable to locate autoload.php file" and now I'm quite at sea.  

Are there any known issues here?  How should I proceed?

 php C:\Users\Lee\.vscode\extensions\junstyle.php-cs-fixer-0.3.10\php-cs-fixer.phar  fix "C:\xxx\code\server\www\Data\phpcsexample.php"
PHP Fatal error:  Uncaught RuntimeException: Unable to locate autoload.php file. in C:\Users\Lee\.vscode\extensions\junstyle.php-cs-fixer-0.3.10\php-cs-fixer.phar:108
Stack trace:
#0 C:\Users\Lee\.vscode\extensions\junstyle.php-cs-fixer-0.3.10\php-cs-fixer.phar(113): {closure}()
#1 {main}
  thrown in C:\Users\Lee\.vscode\extensions\junstyle.php-cs-fixer-0.3.10\php-cs-fixer.phar on line 108

Fatal error: Uncaught RuntimeException: Unable to locate autoload.php file. in C:\Users\Lee\.vscode\extensions\junstyle.php-cs-fixer-0.3.10\php-cs-fixer.phar:108
Stack trace:
#0 C:\Users\Lee\.vscode\extensions\junstyle.php-cs-fixer-0.3.10\php-cs-fixer.phar(113): {closure}()
#1 {main}
  thrown in C:\Users\Lee\.vscode\extensions\junstyle.php-cs-fixer-0.3.10\php-cs-fixer.phar on line 108
  
  
  ETA:
  I've reverted to my previous version of the phar, which also turns out to be 3.24.0.  I speculate this version was installed automatically by the plug-in.  When I tried to run the self updater, I first was forbidden from a rename because the parent directory was read-only (and immutably so, even as Administrator) and then just retrieved a couple of .phar files (for 3.24 and 3.25.1 directly from this github, which led to the errors above.)  I'm left head-scratching and about one version behind.

**Comment by Wirone** [comment]
Sorry to hear that, but I don't have possibility to test on Windows, I don't use VSCode and I have no idea what Junstyle is 🙂. There's no other report about PHAR 3.25.1 being unusable, so I suspect it's your runtime issue only (probably Junstyle plugin bug). Please verify it on your system by [downloading PHAR](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/releases/download/v3.25.1/php-cs-fixer.phar) and using it directly, not through IDE's plugin. We then can be sure it's either phar/Windows, or IDE/plugin issue.

**Reply by codigoergosum** [reply]
Thanks for reviewing this issue; whatever it was, I'm not able to duplicate it now, although yesterday I had in fact downloaded the PHAR as you described before entering this report.  Absent more reports this problem, there doesn't seem to be anything more to do.

Junstyle/vscode-php-cs-fixer is a MS VS Code plugin for PHP-CS-Fixer, mentioned by your docs here: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer#editor-integration

The Visual Studio Marketplace tracks about 850k installs for it, and it appears to be the most recently maintained of such plugins.
